src/main/java/core/basesyntax/Lottery.java

```diff
@@ -0,0 +1,20 @@
+package core.basesyntax;
+
+public class Lottery {
+    private Ball ball;
+    private ColorSupplier color;
+    private NumberBall numberBall;
+
+    public Lottery(Ball ball, ColorSupplier color, NumberBall numberBall) {
+        this.ball = ball;
+        this.color = color;
+        this.numberBall = numberBall;
+    }
+
+    public Ball getRandomBall() {
+        ball.setColor(color.getRandomColor());
+        ball.setNumber(numberBall.getRandomNumber());
+        return ball;
+    }
```

Comment:
Don't create temporary variables if you can directly return value from the method. Also, a new Ball instance should be created and returned each time this method is called to avoid side effects on a single instance.